external/pdal_wrench/main.cpp

```diff
@@ -0,0 +1,327 @@
+/*****************************************************************************
+ *   Copyright (c) 2023, Lutra Consulting Ltd. and Hobu, Inc.                *
+ *                                                                           *
+ *   All rights reserved.                                                    *
+ *                                                                           *
+ *   This program is free software; you can redistribute it and/or modify    *
+ *   it under the terms of the GNU General Public License as published by    *
+ *   the Free Software Foundation; either version 3 of the License, or       *
+ *   (at your option) any later version.                                     *
+ *                                                                           *
+ ****************************************************************************/
+
+/*
+TODO:
+- algs that output point cloud: support multi-copc or single-copc output - as a post-processing step?
+- VPC: overlapping files? do not allow - require tiling
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "alg.hpp"
+#include "vpc.hpp"
+
+extern int runTile(std::vector<std::string> arglist);  // tile/tile.cpp
+
+
+// TODO: make it windows/unicode friendly
+//#ifdef _WIN32
+//int wmain( int argc, wchar_t *argv[ ], wchar_t *envp[ ] )
+//#else
+int main(int argc, char* argv[])
+//#endif
+{
+#if 1
+    if (argc < 2)
+    {
+        std::cerr << "need to specify command:" << std::endl;
+        std::cerr << " - boundary" << std::endl;
+        std::cerr << " - clip" << std::endl;
+        std::cerr << " - density" << std::endl;
+        std::cerr << " - build_vpc" << std::endl;
+        std::cerr << " - info" << std::endl;
+        std::cerr << " - merge" << std::endl;
+        std::cerr << " - thin" << std::endl;
+        std::cerr << " - tile" << std::endl;
+        std::cerr << " - to_raster" << std::endl;
+        std::cerr << " - to_raster_tin" << std::endl;
+        std::cerr << " - to_vector" << std::endl;
+        std::cerr << " - translate" << std::endl;
+        return 1;
+    }
+    std::string cmd = argv[1];
+
+    // TODO: use untwine::fromNative(argv[i])
+    std::vector<std::string> args;
+    for ( int i = 2; i < argc; ++i )
+        args.push_back(argv[i]);
+#elif 0
+    std::string cmd = "density";
+    std::vector<std::string> args;
+    // args.push_back("-i");
+    // args.push_back("/home/martin/qgis/point-cloud-sandbox/data/24-fixed.las");
+    // args.push_back("-o");
+    // args.push_back("/tmp/dens.tif");
+    // args.push_back("-r");
+    // args.push_back("1");
+    //args.push_back("xxx");
+
+    args.push_back("--input=/home/martin/qgis/point-cloud-sandbox/data/merged.copc.laz");
```

Comment:
lots of debug code in here -- will this all be removed?

Reply:
yeah that will be removed, sorry for that. I think we will do a couple of iterations on the pdal_wrench and keep updating it to improve the code robustness, error handling etc. My preference would be to get rid of this code in a later update, but if you prefer, we can do it in this initial PR...

Reply:
Removed in https://github.com/PDAL/wrench/commit/c50d28853b1f9ab7cc236d57cf32115732794da3